metarecord/models/attribute.py

```diff
@@ -64,15 +64,24 @@ def is_free_text(self):
 class AttributeValue(TimeStampedModel, UUIDPrimaryKeyModel):
     attribute = models.ForeignKey(Attribute, verbose_name=_('attribute'), related_name='values')
     value = models.CharField(verbose_name=_('value'), max_length=1024)
+    index = models.PositiveSmallIntegerField(db_index=True)
 
     class Meta:
         verbose_name = _('attribute value')
         verbose_name_plural = _('attribute values')
-        unique_together = ('attribute', 'value')
+        unique_together = (('attribute', 'value'), ('attribute', 'index'))
+        ordering = ('index',)
 
     def __str__(self):
         return self.value
 
+    def save(self, *args, **kwargs):
+        if not self.index:
+            self.index = max(
+                AttributeValue.objects.filter(attribute=self.attribute).values_list('index', flat=True) or [0]
```

Comment:
That's a bit too cryptic. How about this:
```
last_index = AttributeValue.objects.filter(attribute=self.attribute).values_list('index', flat=True).order_by('-index').first()
if not last_index:
    last_index = 0
self.index = last_index + 1
```

Another issue is a race condition, but perhaps we don't need to worry about it here.